pkg/registry/core/service/strategy.go

```diff
@@ -162,6 +167,33 @@ func (serviceStatusStrategy) WarningsOnUpdate(ctx context.Context, obj, old runt
 	return nil
 }
 
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	service, ok := obj.(*api.Service)
+	if !ok {
+		return nil, nil, fmt.Errorf("not a service")
+	}
+	return service.Labels, SelectableFields(service), nil
+}
+
+// Matcher returns a selection predicate for a given label and field selector.
+func Matcher(label labels.Selector, field fields.Selector) pkgstorage.SelectionPredicate {
```

Comment:
Only used in the test - define it there?

Reply:
copied from the other places where this is implemented

```
pkg/registry/core/secret/storage/storage.go:            PredicateFunc:             secret.Matcher,
pkg/registry/core/secret/strategy.go:// Matcher returns a selection predicate for a given label and field selector.
pkg/registry/core/secret/strategy.go:func Matcher(label labels.Selector, field fields.Selector) pkgstorage.SelectionPredicate {
pkg/registry/core/service/strategy_test.go:                     m := Matcher(labels.Everything(), testCase.fieldSelector)
pkg/registry/core/service/strategy.go:// Matcher returns a selection predicate for a given label and field selector.
pkg/registry/core/service/strategy.go:func Matcher(label labels.Selector, field fields.Selector) pkgstorage.SelectionPredicate {
pkg/registry/core/event/storage/storage.go:             PredicateFunc: event.Matcher,
pkg/registry/core/event/strategy.go:// Matcher returns a selection predicate for a given label and field selector.
pkg/registry/core/event/strategy.go:func Matcher(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
pkg/registry/core/configmap/storage/storage.go:         PredicateFunc:             configmap.Matcher,
pkg/registry/core/configmap/strategy.go:// Matcher returns a selection predicate for a given label and field selector.
```

Reply:
And you missed to set the PredicateFunc when creating the store - added a comment above.

Reply:
done